nodl_python/nodl/nodl_types.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY,
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with this program.
+# If not, see <http://www.gnu.org/licenses/>.
+
+
+from rclpy.qos import QoSProfile, QoSPresetProfiles
+
+
+class nodl_action():
+    """Data structure for action entries in NoDL."""
+
+    def __init__(self,
+                 name: str,
+                 msg_type: str,
+                 server: bool = False,
+                 client: bool = False,
+                 qos: QoSProfile = QoSPresetProfiles.ACTION_STATUS_DEFAULT):
+        self.name = name
+        self.type = msg_type
+        self.server = server
+        self.client = client
```

Comment:
Might be worth it to do some sanity check here, 
```python
assert self.server != self.client, "Either a server or a client, not both, not none!"
```

Reply:
Actually, maybe server and client aren't mutually exclusive, `assert self.server or self.client`.

Reply:
Actually, maybe having both set to `false` should trigger a warning rather than an assert :D

Reply:
That should be done in the parser, these datatypes should be passive, essentially just mypy type checked dicts

Reply:
You're right @artivis, that will definitely need to happen, but I agree with @Arnatious that this is a low enough level it's not necessary. These are essentially just PODs, the logic they require will be a level up.